sw/device/tests/crypto/cryptotest/firmware/edn_sca.c

```diff
@@ -0,0 +1,228 @@
+  dif_edn_seed_material_t kEdnKatSeedMaterialReseed = {
+      .len = kEdnKatMaxClen,
+      .data = {reseed[0], reseed[1], reseed[2], reseed[3], reseed[4], reseed[5],
+               reseed[6], reseed[7], reseed[8], reseed[9], reseed[10],
+               reseed[11]}};
+  // Seed material for the EDN generate command.
+  const dif_edn_seed_material_t kEdnKatSeedMaterialGenerate = {
+      .len = 0,
+  };
+
+  dif_edn_auto_params_t edn_params;
+  edn_params.instantiate_cmd.cmd = csrng_cmd_header_build(
+      kCsrngAppCmdInstantiate, kDifCsrngEntropySrcToggleDisable,
+      kEdnKatSeedMaterialInstantiate.len, /*generate_len=*/0);
+  edn_params.instantiate_cmd.seed_material = kEdnKatSeedMaterialInstantiate;
+  edn_params.reseed_cmd.cmd = csrng_cmd_header_build(
+      kCsrngAppCmdReseed, kDifCsrngEntropySrcToggleDisable,
+      kEdnKatSeedMaterialReseed.len,
+      /*generate_len=*/0);
+  edn_params.reseed_cmd.seed_material = kEdnKatSeedMaterialReseed;
+  edn_params.generate_cmd.cmd = csrng_cmd_header_build(
+      kCsrngAppCmdGenerate, kDifCsrngEntropySrcToggleDisable,
+      kEdnKatSeedMaterialGenerate.len,
+      /*generate_len=*/
+      kEdnKatOutputLen / kEdnKatWordsPerBlock);
+
+  edn_params.generate_cmd.seed_material = kEdnKatSeedMaterialGenerate;
+  edn_params.reseed_interval = 32;
+
+  // Disable the entropy complex.
+  TRY(entropy_testutils_stop_all());
+  // Enable ENTROPY_SRC in FIPS mode.
+  TRY(dif_entropy_src_configure(
+      &entropy_src, entropy_testutils_config_default(), kDifToggleEnabled));
+  // Enable CSRNG.
+  TRY(dif_csrng_configure(&csrng));
+  // Enable EDN0 in auto request mode.
+  TRY(dif_edn_set_auto_mode(&edn0, edn_params));
+
+  uint32_t ibex_rnd_data;
+
+  // Capture trace during generation and transportation of random data.
+  sca_set_trigger_high();
+  asm volatile(NOP10);
+  TRY(rv_core_ibex_testutils_get_rnd_data(&rv_core_ibex, kEdnKatTimeout,
+                                          &ibex_rnd_data));
+  sca_set_trigger_low();
+  asm volatile(NOP10);
+
+  return OK_STATUS();
+}
+
+status_t handle_edn_sca_bus_data_batch_fvsr(ujson_t *uj) {
+  // Get seed material.
+  edn_sca_seed_batch_t uj_data;
+  TRY(ujson_deserialize_edn_sca_seed_batch_t(uj, &uj_data));
+
+  bool sample_fixed = true;
+  uint32_t last_value = 0;
+  for (size_t it = 0; it < uj_data.num_iterations; it++) {
+    if (sample_fixed) {
+      // Configure EDN with provided seed material, set trigger and start
+      // generating and fetching data.
+      TRY(config_run_edn(uj_data.init_seed, uj_data.reseed));
+      last_value = uj_data.reseed[11];
+    } else {
+      // Generate random seeds for the EDN configuration.
+      uint32_t init_seed[12];
+      uint32_t reseed[12];
+      generate_random(12, init_seed);
+      generate_random(12, reseed);
+      last_value = reseed[11];
+      // Configure EDN with random seed material, set trigger and start
+      // generating and fetching data.
+      TRY(config_run_edn(init_seed, reseed));
+    }
```

Comment:
Please restructure the code to first generate an array with all random seeds and then loop over the generated array as we do this for AES, SHA3, KMAC. At least of FPGA, we ran into issues where the generation of the random data alone led to a static offset between the fixed and random group just because of the additional time/power consumed by the generation and associated discharge of the power network on the chip / FPGA.